test/jdk/javax/swing/JButton/SpaceKeyActivatesButton.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+
+import static java.util.stream.Collectors.toList;
+
+/*
+ * @test
```

Comment:
@bug ?

And I presume you've verified this using mach 5?

Also do we actually specify that Space generates an ActionEvent on a Button ?
I'm wondering what is the motivation for this test.

Reply:
I couldn't find any bug associated with this, that's why @bug label was not given. I have created a test for [JDK-4659800](https://bugs.openjdk.java.net/browse/JDK-4659800) -> https://git.openjdk.java.net/jdk/pull/7296 which verifies the action of 'Enter' key on focused button, but it was applicable for Windows platform(Windows and Windows Classic LnFs) only. So just thought of creating a test which is applicable for all platforms and in all LnFs. This doc https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/event/ActionEvent.html says that "To invoke an ActionEvent on a Button using the keyboard, use the Space bar", so I created a general test which checks this fact as I couldn't find any existing test related to this use case.
This is well tested using mach5 and the link is attached in the bug.

Reply:
@bug is informational tag and it is not necessary to have product bug to map it. It will help in tracking why this test case was added(especially now when we no more use JBS bug id in test case names).

Reply:
Added @bug label.